app/services/make_an_offer.rb

```diff
@@ -36,4 +38,13 @@ def validate_offer_conditions
     errors.add(:offer_conditions, "has over #{MAX_CONDITIONS_COUNT} elements") if @offer_conditions.count > MAX_CONDITIONS_COUNT
     errors.add(:offer_conditions, "has a condition over #{MAX_CONDITION_LENGTH} chars in length") if @offer_conditions.any? { |c| c.length > MAX_CONDITION_LENGTH }
   end
+
+  def notify_slack
+    provider_name = @application_choice&.course&.provider&.name
```

Comment:
This seems very defensive, I don't think a choice without a course without a provider etc will ever exist.

Reply:
I think it's important the notification code does not raise any errors, lest it rolls back the transaction (as a general thing)